src/game/services/Rules/state-machine/performingBotCropHarvestingState.ts

```diff
@@ -27,8 +28,16 @@ export const performingBotCropHarvestingState: RulesMachineConfig['states'] = {
         const { currentPlayerId } = match
         assertCurrentPlayer(currentPlayerId)
 
-        const plantedCrop =
-          match.table.players[currentPlayerId].field.crops[cropCardIdxToHarvest]
+        if (cropCardIdxToHarvest === undefined) {
+          enqueue.raise({
+            type: MatchEvent.PROMPT_BOT_FOR_TURN_ACTION,
+          })
+          return
+        }
```

Comment:
Avoid early returns in state machine logic. Invert the check on line 31 and move the logic that needs to be skipped if `cropCardIdxToHarvest` is `undefined` into the body of the `if`.